In [0]:
install.packages("wooldridge")
library(wooldridge)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



##### If there is discrimination against minorities, and the appropriate factors have been controlled for, $\beta$<sub>1</sub> will have a positive value, since a positive value will signal discrimination against minorities. Here the value of $\beta$<sub>1</sub> is greater than zero. Therefore it denotes that a white person will have a greater chance of having a loan approved, having other factors controlled. 

In [0]:
attach(loanapp)
m1<-lm(approve~white)
summary(m1)


Call:
lm(formula = approve ~ white)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.90839  0.09161  0.09161  0.09161  0.29221 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  0.70779    0.01824   38.81   <2e-16 ***
white        0.20060    0.01984   10.11   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3201 on 1987 degrees of freedom
Multiple R-squared:  0.04893,	Adjusted R-squared:  0.04845 
F-statistic: 102.2 on 1 and 1987 DF,  p-value: < 2.2e-16


#The Linear regression model

##### **_$approve$_**= 0.7078 + 0.2006 **_$white$_**
#####$n$ =1989
#####$R^{2}$ = 0.049

The coefficient of white denotes that in a sample of 1989 applicants, an application submitted by a white applicant was 20.06% more likely to be approved than non-white applicant. The t-statistic is 10.11 and therefore denotes a large difference. 

In [0]:
m2<- lm(approve ~ white+hrat+obrat+loanprc+unem+male+married+dep+sch+cosign+chist+pubrec+mortlat1+mortlat2+vr)
summary(m2)


Call:
lm(formula = approve ~ white + hrat + obrat + loanprc + unem + 
    male + married + dep + sch + cosign + chist + pubrec + mortlat1 + 
    mortlat2 + vr)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.06482  0.00781  0.06387  0.13673  0.71105 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  0.936731   0.052735  17.763  < 2e-16 ***
white        0.128820   0.019732   6.529 8.44e-11 ***
hrat         0.001833   0.001263   1.451   0.1469    
obrat       -0.005432   0.001102  -4.930 8.92e-07 ***
loanprc     -0.147300   0.037516  -3.926 8.92e-05 ***
unem        -0.007299   0.003198  -2.282   0.0226 *  
male        -0.004144   0.018864  -0.220   0.8261    
married      0.045824   0.016308   2.810   0.0050 ** 
dep         -0.006827   0.006701  -1.019   0.3084    
sch          0.001753   0.016650   0.105   0.9162    
cosign       0.009772   0.041139   0.238   0.8123    
chist        0.133027   0.019263   6.906 6.72e-12 ***
pubrec      -0.241

#####We obtain the coefficient of $white$, $\beta$</sub>1</sub> to be equal to 0.1288 and the $standard$ $error$ to be 0.0197. The coefficinet has fallen by some margin because we now control for factors that affect loan approval rates, and some of these factors clearly differ by race. The race effect is still strong and significant. This may be due to reasons that white people have higher incomes and stronger credit histories. This suggest that there is still evidence of discrimination against nonwhites.

In [0]:
attach(nbasal)
head(nbasal)

The following objects are masked from nbasal (pos = 3):

    age, agesq, allstar, assists, avgmin, black, center, children,
    coll, draft, exper, expersq, forward, games, guard, lwage, marr,
    marrblck, minutes, points, rebounds, wage


The following objects are masked from nbasal (pos = 4):

    age, agesq, allstar, assists, avgmin, black, center, children,
    coll, draft, exper, expersq, forward, games, guard, lwage, marr,
    marrblck, minutes, points, rebounds, wage


The following object is masked from loanapp:

    black




marr wage   exper age coll games minutes guard forward center ⋯ assists draft
1 1    1002.5 4     27  4    77    2867    1     0       0      ⋯ 4.5     19   
2 1    2030.0 5     28  4    78    2789    1     0       0      ⋯ 8.8     28   
3 0     650.0 1     25  4    74    1149    0     0       1      ⋯ 0.2     19   
4 0    2030.0 5     28  4    47    1178    0     1       0      ⋯ 1.5      1   
5 0     755.0 3     24  4    82    2096    1     0       0      ⋯ 2.6     24   
6 0    2014.5 9     31  4    82    1971    0     1       0      ⋯ 1.5      4   
  allstar avgmin   lwage    black children expersq agesq marrblck
1 0       37.23376 6.910252 1     0        16      729   1       
2 0       35.75641 7.615791 1     1        25      784   1       
3 0       15.52703 6.476973 1     0         1      625   0       
4 0       25.06383 7.615791 1     0        25      784   0       
5 0       25.56098 6.626718 1     0         9      576   0       
6 0       24.03658 7.608126 1     0        81      961   0

In [0]:
m3<-lm(points~exper+expersq+guard+forward)
summary(m3)


Call:
lm(formula = points ~ exper + expersq + guard + forward)

Residuals:
    Min      1Q  Median      3Q     Max 
-11.220  -4.268  -1.003   3.444  22.265 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  4.76076    1.17862   4.039 7.03e-05 ***
exper        1.28067    0.32853   3.898 0.000123 ***
expersq     -0.07184    0.02407  -2.985 0.003106 ** 
guard        2.31469    1.00036   2.314 0.021444 *  
forward      1.54457    1.00226   1.541 0.124492    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 5.668 on 264 degrees of freedom
Multiple R-squared:  0.09098,	Adjusted R-squared:  0.07721 
F-statistic: 6.606 on 4 and 264 DF,  p-value: 4.426e-05


#The Linear regression model

##### **_$points$_**= 4.7608 + 1.2807 **_$exper$_** - 0.0718 **_$expersq$_** + 2.3147 **_$guard$_** + 1.5446 **_$forward$_**

#####$n$ =269
#####$R^{2}$ = 0.0772

#####Including all the position dummy variables will be redundat and therefore can result in the dummy variable trap. Each player will fall into one of the three categories and the overall intercept is the intercept for the centers

#####A guard is estimated to score more than a center by 2.3147 points holding experience constant. The t statistic is 2.314, and therefore the difference is is statistically different from zero from the 5% level, against a two-sided alternative

In [0]:
m3<-lm(points~exper+expersq+guard+forward+marr)
summary(m3)


Call:
lm(formula = points ~ exper + expersq + guard + forward + marr)

Residuals:
    Min      1Q  Median      3Q     Max 
-10.874  -4.227  -1.251   3.631  22.412 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  4.70294    1.18174   3.980 8.93e-05 ***
exper        1.23326    0.33421   3.690 0.000273 ***
expersq     -0.07037    0.02416  -2.913 0.003892 ** 
guard        2.28632    1.00172   2.282 0.023265 *  
forward      1.54091    1.00298   1.536 0.125660    
marr         0.58427    0.74040   0.789 0.430751    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 5.672 on 263 degrees of freedom
Multiple R-squared:  0.09313,	Adjusted R-squared:  0.07588 
F-statistic: 5.401 on 5 and 263 DF,  p-value: 9.526e-05


#####When we add marr to the regression, the coefficient is about 0.5843 (se= 0.7404). Therefore, a married player is estimated to score over just 0.5843 points more per game, with other factors (experience and position held constant), but it is not statistically significant from zero, since the p-value is equal to 0.4308. So on the basis of points per game, we cannot conclude that the married players are more productive.

In [0]:
m4<-lm(points~exper+expersq+guard+forward+marr+marr*exper+marr*expersq)
summary(m4)


Call:
lm(formula = points ~ exper + expersq + guard + forward + marr + 
    marr * exper + marr * expersq)

Residuals:
    Min      1Q  Median      3Q     Max 
-10.239  -4.328  -1.067   3.742  22.197 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)   5.81615    1.34878   4.312 2.29e-05 ***
exper         0.70255    0.43405   1.619   0.1067    
expersq      -0.02950    0.03267  -0.903   0.3674    
guard         2.25079    1.00002   2.251   0.0252 *  
forward       1.62915    1.00199   1.626   0.1052    
marr         -2.53750    2.03822  -1.245   0.2143    
exper:marr    1.27965    0.68229   1.876   0.0618 .  
expersq:marr -0.09359    0.04887  -1.915   0.0566 .  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 5.654 on 261 degrees of freedom
Multiple R-squared:  0.1058,	Adjusted R-squared:  0.08184 
F-statistic: 4.413 on 7 and 261 DF,  p-value: 0.0001188


#####Adding the interactions produces different signs (positive and negative) on the three terms involving **_$marr$_**.  The **_$F$_** test for their joint significance, with 3 and 261 _**$df$**_, gives a _**$F$**_ value of 1.44 and _**$p$**_ value of 0.23. Therefore, there is no presence of a strong evidence that marital status has any partial effect on points scored.


In [32]:
m5<-lm(assists~exper+expersq+guard+forward+marr)
summary(m5)


Call:
lm(formula = assists ~ exper + expersq + guard + forward + marr)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.3127 -1.0780 -0.3157  0.6788  8.2488 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -0.225809   0.354904  -0.636  0.52516    
exper        0.443603   0.100372   4.420 1.45e-05 ***
expersq     -0.026726   0.007256  -3.683  0.00028 ***
guard        2.491672   0.300842   8.282 6.19e-15 ***
forward      0.447471   0.301220   1.486  0.13860    
marr         0.321899   0.222359   1.448  0.14891    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.704 on 263 degrees of freedom
Multiple R-squared:  0.3499,	Adjusted R-squared:  0.3375 
F-statistic: 28.31 on 5 and 263 DF,  p-value: < 2.2e-16


#####If we use assists as the dependent variable, the coefficient on marr becomes 0.3219 (se=0.2223). Therefore, it denotes that a married man has almost one-thirs more assist per game, holding experience and position fixed(constant). The p-value, against a two-sided alternative is about 0.1489, which is strong, but not overwhelming, having evidence that married men are more productive when it comes to assist in the games.